src/Compression/CompressionCodecDeflateQpl.cpp

```diff
@@ -364,9 +389,9 @@ void SoftwareCodecDeflateQpl::doDecompressData(const char * source, UInt32 sourc
 }
 
 CompressionCodecDeflateQpl::CompressionCodecDeflateQpl()
-    : hw_codec(std::make_unique<HardwareCodecDeflateQpl>())
-    , sw_codec(std::make_unique<SoftwareCodecDeflateQpl>())
+    : sw_codec(std::make_unique<SoftwareCodecDeflateQpl>())
```

Comment:
Instead of initializing `hw_codec` only in the ctor body but not in the initialization list, could we not simply switch the order of the `hw_codec` and `sw_codec` members in class `CompressionCodecDeflateQpl`?

```cpp
...
private:
    UInt32 getMaxCompressedDataSize(UInt32 uncompressed_size) const override;

    std::unique_ptr<SoftwareCodecDeflateQpl> sw_codec;
    std::unique_ptr<HardwareCodecDeflateQpl> hw_codec;
};
```

Reply:
Done

Reply:
The point of switching the order of members in the class was that we could continue to initialize them in the initialization list:

```cpp
CompressionCodecDeflateQpl::CompressionCodecDeflateQpl()
    : sw_codec(std::make_unique<SoftwareCodecDeflateQpl>())
    , hw_codec(std::make_unique<HardwareCodecDeflateQpl>(sw_codec))
{
    ...
```